Tests/Realm.Tests/Sync/SessionTests.cs

```diff
@@ -100,6 +103,347 @@ public void Session_Error_ShouldPassCorrectSession()
+                Assert.IsTrue(onBeforeTriggered);
+                Assert.IsTrue(onAfterResetTriggered);
+            });
+        }
+
+        [Test]
+        public void Session_SyncError_Handler()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var sessionErrorTriggered = false;
+                var tcs = new TaskCompletionSource<bool>();
+                var config = await GetIntegrationConfigAsync();
+                config.OnSessionError = (sender, e) =>
+                {
+                    Assert.IsInstanceOf<Session>(sender);
+                    Assert.IsInstanceOf<SessionException>(e);
+                    Assert.IsFalse(sessionErrorTriggered);
+                    sessionErrorTriggered = true;
+                    tcs.TrySetResult(true);
+                };
+
+                using var realm = await GetRealmAsync(config);
+                var session = GetSession(realm);
+
+                session.SimulateError(ErrorCode.PermissionDenied, "simulated sync issue");
+
+                await tcs.Task;
+
+                Assert.IsTrue(sessionErrorTriggered);
+            });
+        }
+
+        // after deprecation: temporary test for co-existence of Session.Error and error handling in SyncConfigurationBase
+        [Test]
+        public void Session_ClientReset_Handlers_Coexistence()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var onBeforeTriggered = false;
+                var onAfterTriggered = false;
+                var obsoleteSessionErrorTriggered = false;
+                var tcs = new TaskCompletionSource<bool>();
+                var config = await GetIntegrationConfigAsync();
+                config.ClientResetHandler = new DiscardLocalResetHandler
+                {
+                    OnBeforeReset = (beforeFrozen) =>
+                    {
+                        Assert.IsFalse(onBeforeTriggered);
+                        Assert.IsFalse(onAfterTriggered);
+                        onBeforeTriggered = true;
+                    },
+                    OnAfterReset = (beforeFrozen, after) =>
+                    {
+                        Assert.IsTrue(onBeforeTriggered);
+                        Assert.IsFalse(onAfterTriggered);
+                        onAfterTriggered = true;
+                        tcs.TrySetResult(true);
+                    }
+                };
+
+                using var realm = await GetRealmAsync(config);
+                var session = GetSession(realm);
+
+                // priority is given to the newer appoach in SyncConfigurationBase, so this should never be reached
+#pragma warning disable CS0618 // Type or member is obsolete
+                Session.Error += (sender, e) =>
+                {
+                    obsoleteSessionErrorTriggered = true;
+                    tcs.TrySetResult(true);
+                };
+#pragma warning restore CS0618 // Type or member is obsolete
+
```

Comment:
This approach - and the one in the next test are prone to race conditions - e.g. if the code triggers both the client reset handler callbacks and `Session.Error`, you would not detect this as long as the ordering is correct. It's kind of silly, but you might want to add a `await Task.Delay(1000)` or similar to ensure that `Session.Error` is indeed not triggered.